cppcache/src/TcrEndpoint.cpp

```diff
@@ -504,14 +504,15 @@ void TcrEndpoint::pingServer(ThinClientPoolDM* poolDM) {
   }
 
   if (!m_msgSent && !m_pingSent) {
-    TcrMessagePing* pingMsg = TcrMessage::getPingMessage(m_cacheImpl);
+    TcrMessagePing pingMsg(new DataOutput(m_cacheImpl->createDataOutput()),
```

Comment:
Where is DataOutput deleted? Hopefully, that's not also a leak.

Reply:
This pointer ownership is transfered to the TcrMessage by assigning it to an unique_ptr. I am with you that the way of instantiating the object might lead to think that this is a memory leak, but this is they way is done for most of the TcrMessage implementations :S

Reply:
Yeah - it's not pretty, but it is correct.